src/abstractdataframe/show.jl

```diff
@@ -573,61 +573,193 @@ function _show(io::IO,
+    aux       = names(df)
+    names_len = textwidth.(aux)
+    maxwidth  = max.(9, names_len)
+    names_mat = permutedims(aux)
+    types     = eltype.(eachcol(df))
 
-    nrows = size(df, 1)
-    if rowid !== nothing
-        if size(df, 2) == 0
-            rowid = nothing
-        elseif nrows != 1
-            throw(ArgumentError("rowid may be passed only with a single row data frame"))
-        end
+    # NOTE: If we use `type` here, the time to print the first table is 2x more.
+    # This should be something related to type inference.
+    types_str = compacttype.(eltype.(eachcol(df)), maxwidth) |> permutedims
+
+    crop = :both
+
+    if allcols && allrows
+        crop = :none
+    elseif allcols
+        crop = :vertical
+    elseif allrows
+        crop = :horizontal
     end
+
+    compact_printing::Bool = get(io, :compact, true)
+
+    # This vector stores the column indices that are only floats. In this case,
+    # the printed numbers will be aligned in the decimal point.
+    float_cols = Int[]
+
+    # These vectors contain the number of the row and the padding that must be
+    # applied so that the float number is aligned with the decimal pointing.
+    indices = Vector{Int}[]
+    padding = Vector{Int}[]
+
+    # If the screen is limited, we do not need to process all the numbers.
     dsize = displaysize(io)
-    availableheight = dsize[1] - 7
-    nrowssubset = fld(availableheight, 2)
-    bound = min(nrowssubset - 1, nrows)
-    if allrows || nrows <= availableheight
-        rowindices1 = 1:nrows
-        rowindices2 = 1:0
+    num_rows, num_cols = size(df)
+
+    if !allcols
+        # Given the spacing, there is no way to fit more than W/9 rows of
+        # floating numbers in the screen, where W is the display width.
+        Δc = clamp(div(dsize[2], 9), 0, num_cols)
     else
-        rowindices1 = 1:bound
-        rowindices2 = max(bound + 1, nrows - nrowssubset + 1):nrows
+        Δc = num_cols
     end
-    maxwidths = getmaxwidths(df, io, rowindices1, rowindices2, rowlabel, rowid,
-                             eltypes, buffer, truncstring)
-    width = getprintedwidth(maxwidths)
-    showrows(io, df, rowindices1, rowindices2, maxwidths, splitcols, allcols,
-             rowlabel, summary, eltypes, rowid, buffer, truncstring)
-    return
+
+    if !allrows
+        # Get the maximum number of lines that we can display given the screen
+        # size.
+        Δr = clamp(dsize[1] - 4, 0, num_rows)
+    else
+        Δr = num_rows
+    end
+
+    Δr_lim = cld(Δr, 2)
+
+    # Do not align the numbers if there are more than 500 rows.
+    if Δr ≤ 500
+        @inbounds for i = 1:Δc
+            # TODO: Should we add support to `Union{Nothing, Float}`?
+
+            # Analyze the order of the number to compute the maximum padding
+            # that must be applied to align the numbers at the decimal point.
+            if nonmissingtype(types[i]) <: AbstractFloat
```

Comment:
actually - given the comment above, we will not be padding if `eltype` is `Any` but actually contains only numbers. But I think it is OK.

Reply:
Ok!

Reply:
@nalimilan this is pretty easy. We just need to check if the column is a number and create an alignment vector. This is a vector containing the alignment of each row. Should I do this? What's algorithm should I use to decide the alignment?

Reply:
I'd say apply this to all `<: Union{Number, Missing}` columns?

Reply:
Ok! So I will align to the right all columns that are `<: Union{Number, Missing}` but not `<: Union{AbstractFloat, Missing}`. The latter must be left aligned so that the algorithm that aligns the decimal points work.

Reply:
Done!